src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultStr = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let value = this[i];
+
+      if (value === null || value === undefined) {
+        value = '';
+      }
+
+      resultStr += value;
```

Comment:
you can additionally simplify this part and concatenate string only  if it not `null` and not `undefined`